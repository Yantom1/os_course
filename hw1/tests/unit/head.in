#Invalid args
head
head not_exists
head -5 not_exists
#Normal checks
echo -e "hello1\nhello2\nhello3\nhello4\nhello5\nhello6\nhello7\nhello8\nhello9\nhello10\nhello11" > tmp
head -1 tmp
head -2 tmp
head -3 tmp
head -20 tmp
#Check default behavior (N=10)
head tmp
#Check behavior with redirection
head -20 tmp | grep 1
#Check correct behavior if no newline at end of file
echo -e -n "hello1\nhello2\nhello3" > tmp
head -3 tmp
head -4 tmp
quit
