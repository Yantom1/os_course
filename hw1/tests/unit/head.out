smash> smash error: head: not enough arguments
smash> smash error: open failed: No such file or directory
smash> smash error: open failed: No such file or directory
smash> smash> hello1
smash> hello1
hello2
smash> hello1
hello2
hello3
smash> hello1
hello2
hello3
hello4
hello5
hello6
hello7
hello8
hello9
hello10
hello11
smash> hello1
hello2
hello3
hello4
hello5
hello6
hello7
hello8
hello9
hello10
smash> hello1
hello10
hello11
smash> smash> hello1
hello2
hello3smash> hello1
hello2
hello3smash> 